### ISSUE TITLE
[apex] Avoid @future calls in loops

### ISSUE THREAD

**Issue by JAertgeerts** [pmd/pmd#623]
**Rule Set:** Apex

**Description:** Avoid calling @future methods inside loops. These can cause governor limit exceptions.
 
**Code Sample demonstrating the issue:**

```
public class Random {
  @future
  public void futureMethod() { ... }

  public void badAction() {
    for (Integer i = 0; i < 200; i++) {
      futureMethod(); // not recommended, will break
    }
  }
}
```

**Comment by JAertgeerts** [comment]
@jsotuyod I'll open a PR for this soon, can assign to me.

**Comment by jsotuyod** [comment]
@JAertgeerts looking forward to it!

**Comment by rsoesemann** [comment]
@jsotuyod I recommend to close "feature-request" issues that are not followed by a PR (or ongoing communication) after 2 month.

**Comment by jsotuyod** [comment]
@rsoesemann if the rule idea is correct (makes sense, adds value to users) I'm ok with keeping it open for future reference. Every now and then we find ourselves with some extra time we use to tackle these  kind of issues, which keeps PMD moving forward.

I wouldn't want to reduce PMD development to fixing bugs / false positives / false negatives, and core functionality. On all languages there are plenty of other tools that provide rules we don't (and should!), but until we can unify all these into a **Single Grand Rule Checklist™** on which to work forward feature parity, I'd rather have these stand alone issues than nothing.

**Comment by noerremark** [comment]
@rsoesemann - Is this Issue still valid?

**Comment by rsoesemann** [comment]
@noerremark yes if you can build it. Will be really hard. Think of @future from other classes.
